Fix compilation error in regexp lang guide (#6703)
ballerina-platform/ballerina-dev-website: swan-lake/learn-the-language/distinctive-language-features/advanced-general-purpose-language-features.md

import ballerina/io;
import ballerina/lang.regexp;

public function main() {
    string[] names = re `,`.split("Bob,Frank,Will,Jack"); // ["Bob","Frank","Will","Jack"]
    
    int patternCount = re `[bB].tt[a-z]*`.findAll("Butter was bought by Betty but the butter was bitter.").length(); // 4

    string result = re `0+`.replaceAll("10010011", "*"); // 1*1*11

    // Extract the username and domain name from an email
    regexp:Groups? emailGroups = re `([a-z]+)@([a-z]+\.[a-z]{2,})`.findGroups("bob@example.net");
    if emailGroups is regexp:Groups {
        regexp:Span username = <regexp:Span>emailGroups[1];

>         // Prints the matched substring, start index, and end index

Reviewer: What if changing to something like this
```
        // Prints the username and its, starting and ending indexes of the email string.
```

Author: ```
Prints the username, start index and end index of the username substring
```
Above is what we had previously and we derived the current version from it.

Author: @pcnfernando wdyt?

        io:println(string `substring: ${username.substring()} start: ${username.startIndex} end: ${username.endIndex}`);

>     }

Reviewer: Wouldn't it be better to specify that the domain name part would be in the 3rd group since it is not being printed here?
```
// <regexp:Span>emailGroups[1] - username
// <regexp:Span>emailGroups[2] - domain name
```

import ballerina/io;
import ballerina/lang.regexp;

public function main() {
    string[] names = re `,`.split("Bob,Frank,Will,Jack"); // ["Bob","Frank","Will","Jack"]
    
    int patternCount = re `[bB].tt[a-z]*`.findAll("Butter was bought by Betty but the butter was bitter.").length(); // 4

    string result = re `0+`.replaceAll("10010011", "*"); // 1*1*11

    // Extract the username and domain name from an email
    regexp:Groups? emailGroups = re `([a-z]+)@([a-z]+\.[a-z]{2,})`.findGroups("bob@example.net");
    if emailGroups is regexp:Groups {
        regexp:Span username = <regexp:Span>emailGroups[1];

>         // Prints the username and its, starting and ending

Reviewer 2: ```suggestion
        // Prints matched substring and its starting and ending indexes
```

>         // Prints the username and its, starting and ending

Reviewer: Wouldn't it be much better if changed to: 
```suggestion
        // Prints the matched substring with its starting and ending indexes
```

        io:println(string `substring: ${username.substring()} start: ${username.startIndex} end: ${username.endIndex}`);

        regexp:Span domain = <regexp:Span>emailGroups[2];

>         // Prints the domain and its, starting and ending

Reviewer 2: ```suggestion
```

string result = re `0+`.replaceAll("10010011", "*"); // 1*1*11

// Extract the username and domain name from an email

> public function main() {

Reviewer 2: Have the whole sample inside the main

    regexp:Groups? emailGroups = re `([a-z]+)@([a-z]+\.[a-z]{2,})`.findGroups("bob@example.net");
    if emailGroups is regexp:Groups {

>         regexp:Span username = <regexp:Span>emailGroups[1];

Reviewer 3: ```suggestion
        regexp:Span username = <regexp:Span> emailGroups[1];
```

>         regexp:Span username = <regexp:Span> emailGroups[1];

Reviewer 2: ```suggestion
        regexp:Span username = <regexp:Span>emailGroups[1];
```

>         // Prints the username, start index and end index of the username substring

Reviewer 2: ```suggestion
        // Prints the username, start index, and end index of the matched substring
```
